plugins/logging/file

```diff
@@ -230,7 +230,9 @@ sub maybe_reopen {
         unless ($self->open_log($new_output, $transaction)) {
             return undef;
         }
-        $transaction->notes('file-reopened-this-session', 1);
+        if (UNIVERSAL::can($transaction,'isa')) {
```

Comment:
$transaction->notes is called in a few other places. I only added the test around the ones that were causing problems.